### DISCUSSION TITLE
I need path of web to be in hebrow langwish. and i dont know how to do that thank! #

### DISCUSSION THREAD

**Discussion by edanTavory** [vercel/next.js#38760] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

I need path of web to be in hebrow langwish. and i dont know how to do that thank! #

### What browser are you using? (if relevant)

chrome

### Describe the Bug

I need path of web to be in hebrow langwish. and i dont know how to do that thank! #

### Expected Behavior

path on hebrow langwish

### Link to reproduction

i dont have

### To Reproduce

I need path of web to be in hebrow langwish. and i dont know how to do that thank! #

**Comment by Schniz** [comment]
Hey Edan! I'll write it down in Hebrew for your convenience, but English will follow for visibility! 🙏 

---

מה קורה עידן? תוכל בבקשה אולי לספר מה אתה מצפה שיקרה? מה אתה מנסה לעשות? נראה ש[חלקים דינמיים בpath](https://nextjs.org/docs/routing/dynamic-routes) כן עובדים בעברית, אבל אם הקובץ הוא בעברית אז Next.js פשוט לא מזהה אותו. אם אתה מתכוון לשם קובץ בעברית אז זה באמת לא עובד וזה באג, אבל מניסיוני (וגם לדעתי) זה לא מאוד שכיח שיהיה pathname בעברית באתר (תסתכל לדוגמה על אתרי החדשות הישראלים-הכתובות באנגלית), ועוד יותר לא שכיח לראות קובץ בעברית בפרויקט תכנותי.

בכל מקרה הייתי שמח לדעת מה אתה מנסה לעשות ומה המטרה כדי שנוכל לעזור בצורה הכי טובה שאפשר 😃 

כנראה הבאג של שמות הקבצים בשפה אחרת רלוונטי גם למערכות בערבית או עם א"ב קירילי (רוסית וכו')

---

Can you please elaborate more about what you're trying to achieve? [Dynamic URL segments](https://nextjs.org/docs/routing/dynamic-routes) work with Hebrew letters (I just confirmed), but page files seems to not able to have Hebrew letters in them (aka `pages/אהלן.js` did not work for me), so if the latter is what you're aiming then it is a bug. However, IMO and in practice, it's fairly uncommon to have paths in Hebrew and/or filenames in coding projects that are in Hebrew 😨 so if you can share a bit of what you're trying to do, it will be very helpful 😃 

I guess the same issue could be related for Arabic/Cyrillic filenames, and I'm not entirely sure about news sites in languages with a different alphabet do their routing, so maybe that's a bug we should fix anyway.

**Comment by edanTavory** [comment]
היי תודה רבה על החזרה!
אני רואה שמות של דפים בעברית, בדינאמי זה עובד אבל אני רוצה שכול דפי האתר
יהיו בעברית . בחרתי בnext js
בשביל seo ומנסיוני שם הדף משפיע על קידום אורגני האם יש פתרון? אולי חלופה
לראוטר אחר?
בתאריך יום ב׳, 18 ביולי 2022 ב-15:41 מאת Gal Schlezinger <
***@***.***>:

**Comment by Schniz** [comment]
אני לא מומחה ב-SEO אבל אני חושב שכותרת/meta tags וכל הדברים מסביב משפיעים יותר מאשר הpathname. הpathname בעברית יכול לבלבל המון כלים (במיוחד כלי פיתוח כמו שראית פה עם Next.js), אז אישית, הייתי נמנע מזה. הזכרתי את אתרי החדשות (ynet, n12) בתור דוגמה, ששם SEO זה משהו מאוד חשוב. אם מסתכלים על הכתובות שלהם, הן אפילו לא מכילות שום דבר שקשור לכתבה אלא מה שנראה כמו ID של הכתבה: אבל ה-meta tags שעוזרים למנועי החיפוש להבין את תוכן העמוד בצורה טובה.

תדמיין שיש לך אתר עם i18n עם כל מיני שפות, כנראה שתשמור על היררכיית הקבצים והכתובות להיות זהה בין כל השפות, וזה לא אמור להשפיע על ה-SEO. כלומר לא תראה את הדבר הבא לדעתי ברוב המערכות:

* `/en/ambulances/`
* `/de/krankenwagen/`
* `/he/אמבולנסים/`

אלא משהו בסגנון של:

* `/en/ambulances/`
* `/de/ambulances/`
* `/he/ambulances/`

אז אני חושב שהקבצים יכולים להיות באנגלית וזה לא ישפיע אבל אני לא מומחה בנושא הזה. בכל מקרה, אני חושב שזה כן באג והוא כן יטופל, אבל זה פשוט משהו ממש לא שכיח למיטב היכרותי

---

עריכה: אני מוסיף לפה את התגובה שלי עם הworkaround

```js
/** @type {import('next').NextConfig} */
const config = {
  rewrites() {
    return [
      { source: `/${encodeURIComponent("אהלן")}`, destination: '/welcome' }
    ]
  }
}

module.exports = config;
```

זה יגרום לבקשה ל`/אהלן` להגיע ל`/welcome` ולרנדר את welcome.js

**Reply by edanTavory** [reply]
תודה רבה! יש לך מושג מתי הבאג יטופל

**Reply by Schniz** [reply]
אני חושב שאפשר לעקוב אחרי:

* https://github.com/vercel/next.js/issues/37373
* https://github.com/vercel/next.js/issues/10084

כרגע אני לא רואה שמישהו עובד על זה. בכל מקרה יש workaround והוא [להשתמש בrewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) וככה תקבל את "הטוב מכל העולמות". הנה דוגמה ל next.config.js לדוגמה:

```js
/** @type {import('next').NextConfig} */
const config = {
  rewrites() {
    return [
      { source: `/${encodeURIComponent("אהלן")}`, destination: '/welcome' }
    ]
  }
}

module.exports = config;
```

זה יגרום לבקשה ל`/אהלן` להגיע ל`/welcome` ולרנדר את welcome.js

**Reply by edanTavory** [reply]
זאת אומרת שזה יוצג /אהלן בדפדפן?

**Reply by Schniz** [reply]
כן, וירנדר את /welcome מאחורי הקלעים 😃 

![image](https://user-images.githubusercontent.com/2054772/179532987-ec2ddbf5-72ee-430c-8a34-e1b535bf72d4.png)

**Comment by edanTavory** [comment]
אוקי תודה רבה רבה !!

‫בתאריך יום ב׳, 18 ביולי 2022 ב-17:24 מאת ‪Gal Schlezinger‬‏ <‪
***@***.***‬‏>:‬

**Reply by edanTavory** [reply]
גל דבר נוסף, יש לי דף articels  בו בעצם ותחתיו אני שם את כול המאמרים אני מצליח לשנות את שמות הדפים אך דפים ברמה שלישית מתחת  לא עובד לי. home->articels->heat_pump_for_under_floor_heating
// next.config.js
module.exports = {
  rewrites() {
    return [
      { source: `/${encodeURIComponent("חימום-תת-רצפתי-במים/")}`, destination: '/under_floor_water_heating' },
      { source: `/${encodeURIComponent("חימום-תת-רצפתי-חשמלי")}`, destination: '/under_floor_electric_heating' },
      { source: `/${encodeURIComponent("מאמרים")}`, destination: '/articels' },
      { source: `/${encodeURIComponent("משאבות")}`, destination: '/heat_pump_for_under_floor_heating'},
    ]
  },
  images: {
    domains: ['res.cloudinary.com'],
  },
}

**Reply by Schniz** [reply]
תוכל להשתמש ב code block בגיטהאב כדי שיהיה אפשר לקרוא את הקוד? וגם: מה אתה מנסה להשיג ומה קורה בפועל?

**Reply by edanTavory** [reply]
// next.config.js
module.exports = {
  rewrites() {
    return [
      { source: `/${encodeURIComponent("חימום-תת-רצפתי-במים/")}`, destination: '/under_floor_water_heating' },
      { source: `/${encodeURIComponent("חימום-תת-רצפתי-חשמלי")}`, destination: '/under_floor_electric_heating' },
      { source: `/${encodeURIComponent("מאמרים")}`, destination: '/articels' },
      { source: `/${encodeURIComponent("משאבות")}`, destination: '/heat_pump_for_under_floor_heating'},
     
    ]
  },
  images: {
    domains: ['res.cloudinary.com'],
  },
}
#

**Reply by edanTavory** [reply]
דפים מתחת לדף הבית אני מצליח לשנות שם אך לדוגמא יש לי דף מאמרים מתחת לדף הבית שבו אני רוצה לשים מאמרים 
http://localhost:3000/משאבות/מאמרים
ברמה השלישית אני לא מצליח

**Reply by edanTavory** [reply]
היי גל יש לך דרך כזאת על מנת לבצע את זה?

**Reply by edanTavory** [reply]
??

**Reply by Schniz** [reply]
אני במילואים לכמה שבועות אז זו תהיה בעיה לעזור עכשיו 🥲🥲

**Comment by edanTavory** [comment]
טוב תודה מקווה לפתור אתזה חבל שאין משהו מבונה

בתאריך יום ב׳, 25 ביולי 2022 ב-9:05 מאת Gal Schlezinger <
***@***.***>:
